engine/src/test/java/io/camunda/zeebe/engine/processing/timer/TimerStartEventTest.java

```diff
@@ -877,4 +877,298 @@ public void shouldTriggerOnlyTimerStartEvent() {
+
+    final TimerRecordValue secondTimerRecord =
+        RecordingExporter.timerRecords(TimerIntent.TRIGGERED)
+            .withProcessDefinitionKey(secondDeployment.getProcessDefinitionKey())
+            .getFirst()
+            .getValue();
+
+    Assertions.assertThat(secondTimerRecord)
+        .hasDueDate(secondDueDate)
+        .hasTargetElementId("start_2")
+        .hasElementInstanceKey(TimerInstance.NO_ELEMENT_INSTANCE);
+
+    assertThat(
+            RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATED)
+                .withElementId("process_2")
+                .withProcessDefinitionKey(secondDeployment.getProcessDefinitionKey())
+                .exists())
+        .isTrue();
+  }
+
+  @Test
+  public void shouldTriggerOnlyTwice() {
+    // given
+    final ZonedDateTime start =
+        ZonedDateTime.ofInstant(
+                Instant.ofEpochMilli(engine.getClock().getCurrentTimeInMillis()),
+                ZoneId.systemDefault())
+            .plusSeconds(10);
+
+    final BpmnModelInstance firstModel =
+        Bpmn.createExecutableProcess("process_1")
+            .startEvent("start_1")
+            .timerWithCycle(String.format("R2/%s/PT10S", start))
+            .endEvent("end_1")
+            .done();
+
+    final BpmnModelInstance secondModel =
+        Bpmn.createExecutableProcess("process_2")
+            .startEvent("start_2")
+            .timerWithCycle(String.format("R3/%s/PT10S", start))
+            .endEvent("end_2")
+            .done();
+
+    final var firstDeployment =
+        engine
+            .deployment()
+            .withXmlResource(firstModel)
+            .deploy()
+            .getValue()
+            .getProcessesMetadata()
+            .get(0);
+
+    final var secondDeployment =
+        engine
+            .deployment()
+            .withXmlResource(secondModel)
+            .deploy()
+            .getValue()
+            .getProcessesMetadata()
+            .get(0);
+    final long firstDeploymentProcessDefinitionKey = firstDeployment.getProcessDefinitionKey();
+    final long secondDeploymentProcessDefinitionKey = secondDeployment.getProcessDefinitionKey();
+
+    // when
+    engine.increaseTime(Duration.ofSeconds(25));
+
+    // then
+    assertThat(
+            RecordingExporter.timerRecords(TimerIntent.TRIGGERED)
+                .withProcessDefinitionKey(firstDeploymentProcessDefinitionKey)
+                .exists())
+        .isTrue();
+
+    assertThat(
+            RecordingExporter.timerRecords(TimerIntent.TRIGGERED)
+                .withProcessDefinitionKey(secondDeploymentProcessDefinitionKey)
+                .exists())
+        .isTrue();
+
+    // when
+    engine.increaseTime(Duration.ofSeconds(5));
+
+    // then
+    assertThat(
+            RecordingExporter.timerRecords(TimerIntent.TRIGGERED)
+                .withProcessDefinitionKey(firstDeploymentProcessDefinitionKey)
+                .count())
+        .isEqualTo(2);
```

Comment:
:x: We should avoid using `.count()` without a limit. The test waits always 5 seconds on `.count()`.

Instead of waiting (25 seconds), let's verify the triggered events only. We can assume that the ordering of the events is deterministic (i.e. always `process_1` before `process_2`).

To verify that both processes are triggered twice:

```
assertThat(RecordingExporter.timerRecords(TimerIntent.TRIGGERED).limit(4))
        .extracting(record -> record.getValue().getProcessDefinitionKey())
        .containsExactly(
            firstDeploymentProcessDefinitionKey,
            secondDeploymentProcessDefinitionKey,
            firstDeploymentProcessDefinitionKey,
            secondDeploymentProcessDefinitionKey);
``` 

To verify that the second process is triggered three times:

```
assertThat(RecordingExporter.timerRecords(TimerIntent.TRIGGERED).limit(5))
        .extracting(record -> record.getValue().getProcessDefinitionKey())
        .containsExactly(
            firstDeploymentProcessDefinitionKey,
            secondDeploymentProcessDefinitionKey,
            firstDeploymentProcessDefinitionKey,
            secondDeploymentProcessDefinitionKey,
            secondDeploymentProcessDefinitionKey);
```

Under the assumption of a deterministic order, we can verify that the first process is not triggered by checking that the second process is triggered a third time without the first process in between.